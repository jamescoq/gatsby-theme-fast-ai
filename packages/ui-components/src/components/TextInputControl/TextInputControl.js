import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';
import MaskedInput from 'react-text-mask';

import FormControl from '../FormControl';

const TextInputControl = forwardRef(
	({ children, className, mask, maskOptions, type = 'text', ...rest }, ref) => {
		const isMasked = maskOptions || mask;

		const component = isMasked ? MaskedInput : 'input';
		const maskProps = isMasked ? { mask, ...maskOptions } : {};

		return (
			<FormControl
				className={cx(className)}
				component={component}
				type={type}
				ref={ref}
				{...maskProps}
				{...rest}
			>
				{children}
			</FormControl>
		);
	}
);

TextInputControl.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	mask: PropTypes.oneOfType([PropTypes.array, PropTypes.func]),
	maskOptions: PropTypes.shape({
		guide: PropTypes.bool,
		placeholderChar: PropTypes.string,
		keepCharPositions: PropTypes.bool,
		pipe: PropTypes.func,
		showMask: PropTypes.bool,
	}),
	/** HTML type attribute of the input. */
	type: PropTypes.string,
};

export default TextInputControl;
