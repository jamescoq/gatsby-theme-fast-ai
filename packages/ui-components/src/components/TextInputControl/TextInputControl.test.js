import React from 'react';
import { shallow } from 'enzyme';

import TextInputControl from './TextInputControl';

describe('<TextInputControl />', () => {
	const requiredProps = { name: 'foo' };

	it('shallow', () => {
		const wrapper = shallow(<TextInputControl {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
