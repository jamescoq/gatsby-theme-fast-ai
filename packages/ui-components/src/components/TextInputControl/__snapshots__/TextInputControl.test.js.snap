// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputControl /> shallow 1`] = `
<FormControl
  className=""
  component="input"
  name="foo"
  onBlur={[Function]}
  onFocus={[Function]}
  type="text"
/>
`;
